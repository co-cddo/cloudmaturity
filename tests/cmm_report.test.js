const m = require("../src/assets/cmm_report.js");

localStorage.setItem(
  "cmm",
  JSON.stringify({
    cost: {
      "4475d1b06a436057e4631e18a6342600d8758ab9bac4fb184096d0902666c15b": "1",
      c75727f46f0009946cb138363efd481a34d7cc9a5fe2e6dd51020c1f1fe33519: "3",
      "54f5219f07624b77f9f39ee02914a180349da2deab195f7463255931dcd711f3": "",
      "4108c7cd54ef2a0937137bb7e4eb0d6e1d94d0f2975fd05c4c62b19a27bdc0b8": "",
      "91594c2ad1ed87cb7174242e43315d5fe339a23cf0efc5817c3b083350d1bfde": "",
    },
    intro: {
      department: "",
    },
    data: {
      e16b63630921b1ae801ad94a211175f84d2b1bf51840dd0150ab26ed5930b722: "1",
      "742afc97a9484479c47d2d7937bca8022e9b3a45c4214d1f0fee48d8f692f67c": "",
    },
    governance: {
      "428af0fd388228b9dc3b511212e89cf5394bc330d7be220b9a843f2ffc893155": "3",
      "2abc2949a4f3855240604edad217b14f99b23e5c0217598953bbeeb152fa8239": "",
      "1e7ce5e28d81a52fcdf78a4204b35464a100475ed1305023bf618db4c03c52d6": "",
      "0ed23060ed9f404aadaa9ca80e59cdbefadcb0c3d8291d19665f1a12aaeba9a1": "",
      "64ca6af1436bbc1a30c90368500c2b9eb66d0a9ee4a26ac4a014bdd1b62a89d2": "",
      "2b89fa3f052019c64f427e9e51433ff694e3c9b75b15c3a8ff13a161ed6af0a4": "",
      b2f88fee1d53101265aab580c60d2a11c0d15bf780e5619213f6dfbb3191568d: "",
    },
    operations: {
      "6ff06b436c5961bcfe384e5aa6e0bd2e88829bdbede0cfa64a174b4e7299f9e7": "1",
      "43412fd428e8e6c73ab2f7e0c783ca05a4bae3994a4ca876a3316c40ffe5d6e8": "",
      "06fdcd22135a53b620c48ca3f88c9cb57848501426d2109d0e5f183d09102608": "",
      cbcaa211226f476fedd7437ab507d6581913c2d10ed4e5d49ff0b6c734e2a87d: "",
      b4c63c988a8b409ee09425a72f2bf27e70af66e20aa0d73bcca88656df8976f9: "",
      "10bb4bca2281708f06565d0b49e89a3119f2fdd16ad3dae58d97e4abc5cec267": "",
      "3cc27d02f50317fca2c2adeb9b6b1315ac6154d9733d837b7185648e458fcdc7": "",
      "0ccb0dba2d3d18d2561b544a44c3ed1a1b008323f6ef899aba6fc08f83d2d229": "",
      "3381eb521f86f1069789755ad8445c4f27452f86925a5b8e7fa7d0999e5bce01": "",
    },
    people: {
      cca9bd3d218d4ee68e7c5f9b0c169c3a11090daac67d513cce3ebc097ba44871: "1",
      "9fbe1f14fb304b772039949f8878078aafe3ad69f016c1e8e5473b886178bdea": "",
      d981072ed254f3a9bd82f1a74fc4e0e07b97fd8f95ecff297361c366a0a3f974: "",
      b6bbcda28aae3a13df88c1c03d41ee6f8d699b81bb67ce9c87269f4c9899d946: "",
      "2759c3353516e2d7546fc6a69a44c3670bdd5b056a7ec036fa521108651d0732": "",
      "6c0ac0da237bdcd1d37baf11d78752bffe6b4b5dcd518411ee78e9ed6d434014": "",
      "1f1093037bd4bcdcd620204270b5d32827d863424bda3eb2da6b4528afadb773": "",
      ea5829e5c806f349f9057560079c873cb126dbb7c32e1e6e7d71b50f4cda3b01: "",
    },
    security: {
      d2428f0c7f2247fc1c4d95da699a4be5354885a3449f6f6d3d3ec099f524867a: "1",
      "00e7978db3091562dfa0160970e048ad89af7f728537fe077e5d211dc0b889f9": "",
      f83c9aa69398821a82046c3ddde00ab0b7b50fe6156dc3864f4227f29079e84c: "",
      "113dce19f5fbe074d6657ffdc352b89860182f4a8ecae20d71839cb35234b213": "",
      "136919daf282702358121ab95ddc41331483c0e94ccd1448fbeb5cea7c0822d5": "",
      "934297116ea36703a8a7481c42870ba087bbf4d9ef948ab717521c5eb64a1ee8": "",
      "736c3378870478b0d23ef422d3896d79a26f1f84d1b4329bb151e9ee4377456b": "",
      "12e525b040e294983a4f7fd36ef896f6969b0155966ad662b0a333890042b96b": "",
      "5bcfb68a2e4155da71d1697a9a64a83ea4d114b1b8ef1959ab3c1e39d821b48d": "",
      a0b8c1d56928976ef9ff980033f34eab84a5e622d79980927d8beded0f450de0: "",
      "77ea7d0b6e7d049ff90d3381b6ab0fdf7b3f22e6d1eee487bf586f049b675bae": "",
      "4bb8329b6820c6e405a7669d4a83f72935d79a6bfc9fad808a79dccfcae6c796": "",
      "047e43ae761d64c0c53fa740305f984d5d400ea585887714644d7e5969f58aeb": "",
      "624b94fef90405ecb12c024e345581889452997c08c187d077b1ca6664f00376": "",
    },
    tech: {
      "25c91f48b584434e26b736dd316045df698eee3764a19d63b7f03a6057ded9c9": "1",
      "5e3876281754737c7564fbdecb11909835917b5b70c2b9a465847a15c313b5e3": "",
    },
  }),
);

describe("getCategories", () => {
  it("should return an array of categories", () =>
    expect(m.getCategories()).toBeInstanceOf(Array));
});
describe("getScorePercent", () => {
  it("should return a percent score", () =>
    expect(m.getScorePercent("tech")).toMatchSnapshot());
});
describe("getScore", () => {
  it("should return the score", () =>
    expect(m.getScore("tech")).toMatchSnapshot());
});
describe("getScoreMax", () => {
  it("should return max score", () =>
    expect(m.getScoreMax("tech")).toMatchSnapshot());
});
describe("getCategoryTitle", () => {
  it("should return the title", () =>
    expect(m.getCategoryTitle("tech")).toBe("Technology"));
});
describe("createEl", () => {
  it("Should create a DOM element in a parent one and return it", () => {
    const parent = document.createElement("body");
    expect(m.createEl("div", parent)).toBeInstanceOf(HTMLDivElement);
  });
});
describe("addRule", () => {
  it("Should add the CSS rule to the document", () => {
    m.addRule("div", { background: "blue" });
    const el = document.createElement("div");
    expect(getComputedStyle(el).background).toBe("blue");
  });
});
describe("getScoreTable", () => {
  it("should return the table", () =>
    expect(m.getScoreTable("tech")).toMatchSnapshot());
});
describe("renderReport", () => {
  it("Should add the table to the document", () => {
    let el = document.createElement("div");
    el.id = "report-goes-here";
    document.body.appendChild(el);
    m.renderReport();
    expect(document.getElementById("report_radar").outerHTML).toContain(
      "table",
    );
  });
});
