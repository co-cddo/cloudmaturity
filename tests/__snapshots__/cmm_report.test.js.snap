// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getScore should return the score 1`] = `2`;

exports[`getScoreMax should return max score 1`] = `10`;

exports[`getScorePercent should return a percent score 1`] = `0.2`;

exports[`getScoreTable should return the table 1`] = `
<table
  class="radar chaarts"
  id="report_radar"
  style="--1: 0.28; --2: 0.2; --3: 0.2571428571428571; --4: 0.2; --5: 0.2; --6: 0.2; --7: 0.2; --items: 7; --8: var(--1);"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Cost & Sustainability
      </th>
      <th
        scope="col"
      >
        Data
      </th>
      <th
        scope="col"
      >
        Governance
      </th>
      <th
        scope="col"
      >
        Operations
      </th>
      <th
        scope="col"
      >
        People
      </th>
      <th
        scope="col"
      >
        Security
      </th>
      <th
        scope="col"
      >
        Technology
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <span>
          0.28
        </span>
      </td>
      <td>
        <span>
          0.2
        </span>
      </td>
      <td>
        <span>
          0.2571428571428571
        </span>
      </td>
      <td>
        <span>
          0.2
        </span>
      </td>
      <td>
        <span>
          0.2
        </span>
      </td>
      <td>
        <span>
          0.2
        </span>
      </td>
      <td>
        <span>
          0.2
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
